src/txdb.cpp

```diff
@@ -145,15 +133,18 @@ bool CCoinsViewDB::GetStats(CCoinsStats& stats) const
                         nTotalAmount += out.nValue;
                     }
                 }
-                stats.nSerializedSize += 32 + slValue.size();
+                stats.nSerializedSize += 32 + pcursor->GetKeySize();
                 ss << VARINT(0);
+            } else {
+                return error("CCoinsViewDB::GetStats() : unable to read value");
             }
-            pcursor->Next();
-        } catch (const std::exception& e) {
-            return error("%s : Deserialize or I/O error - %s", __func__, e.what());
+        } else {
+            break;
         }
+        pcursor->Next();
     }
-    stats.nHeight = mapBlockIndex.find(GetBestBlock())->second->nHeight;
+
+    stats.nHeight = WITH_LOCK(cs_main, return mapBlockIndex.find(stats.hashBlock)->second->nHeight;);
```

Comment:
Do we really need the lock here?

Reply:
Since we are using `find` it is probably better to have it.
But now we can remove the one in `gettxoutsetinfo` RPC (ref: bitcoin/bitcoin#6290)